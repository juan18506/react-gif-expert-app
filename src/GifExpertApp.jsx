import { useState } from 'react';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([ 'Action', 'Drama' ]);

  const onAddCategory = () => setCategories([ ...categories, 'Fantasy' ]);

  return (
    <>
      <h1>GifExpertApp</h1>

      <button onClick={ onAddCategory }>Add Category</button>

      <ol>
        { 
          categories.map( category => (
            <li key={ category }>{ category }</li>
          ))
        }
      </ol>
    </>
  );
}
