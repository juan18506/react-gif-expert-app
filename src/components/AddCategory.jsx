import { useState } from 'react';

export const AddCategory = ({ updateCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue( target.value );
  }

  const onSubmit = ( event ) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();
    if ( newInputValue.length <= 1 ) return;

    updateCategories( newInputValue );
    setInputValue('');
  }

  return (
    <form onSubmit={ onSubmit }>
      <input
        type="text"
        placeholder="Search gifs"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  );
}
