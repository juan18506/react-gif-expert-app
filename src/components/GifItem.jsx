import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
  return (
    <li className="card">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </li>
  );
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url  : PropTypes.string.isRequired,
}