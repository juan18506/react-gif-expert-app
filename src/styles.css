* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
}

body {
  padding: 3.75rem;
}

#root,
section {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  gap: 1em;
}

form {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

input {
  width: 80%;
  padding: .625rem .9375rem;
  background-color: white;
  border-radius: .3125rem;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
}

h1 {
  font-size: 3rem;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: .3125rem;
}

.card-grid {
  padding: 0;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  gap: 1em;
}

.card {
  height: 0%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  align-items: center;
  border-radius: .625rem;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  overflow: hidden;
  background-color: white;
}

.card p {
  padding: .3125rem 1.25rem 0px 1.25rem;
  margin-top: .3125rem;
  flex: 1;
  text-align: center;
  font-size: 1.5rem;
  background-color: white;
}

.card img {
  width: 100%;
}
