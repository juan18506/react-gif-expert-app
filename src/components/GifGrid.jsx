import { GifItem, Loading } from '../components';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category );

  return (
    <section>
      <h3>{ category }</h3>

      {
        isLoading && ( <Loading /> )
      }
      
      <ul className="card-grid">
        {
          images.map(( image ) => (
            <GifItem 
              key={ image.id }
              { ...image }
            />
          ))
        }
      </ul>
    </section>
  );
}
