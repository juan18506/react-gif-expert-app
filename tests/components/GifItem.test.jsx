import { describe, it, expect } from 'vitest';
import { screen, render } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Tests in <GifItem />', () => {
  const title = 'A test title';
  const url   = 'https://test-url.com/testing';

  it('should match with snapshot', () => {
    const { container } = render( 
      <GifItem title={ title } url={ url } /> 
    );
    expect( container ).toMatchSnapshot();
  });

  it('should show img with correct URL and ALT', () => {
    render( <GifItem title={ title } url={ url } /> );
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );
  });

  it('should show title', () => {
    render( <GifItem title={ title } url={ url } /> );
    expect( screen.getByText(title) ).toBeTruthy();
  });
});
