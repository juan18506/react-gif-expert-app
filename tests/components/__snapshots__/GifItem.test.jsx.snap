// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests in <GifItem /> > should match with snapshot 1`] = `
<div>
  <li
    class="card"
  >
    <img
      alt="A test title"
      src="https://test-url.com/testing"
    />
    <p>
      A test title
    </p>
  </li>
</div>
`;
